src/ui/ozone/platform/wayland/ivi_surface_wrapper.cc

```diff
@@ -86,8 +88,18 @@ void IviSurfaceWrapper::SetWindowGeometry(const gfx::Rect& bounds) {
 void IviSurfaceWrapper::HandleConfigure(void* data,
                               struct ivi_surface* shell_surface,
                               int32_t width,
-                              int32_t height) { 
+                              int32_t height) {
   IviSurfaceWrapper* surface = static_cast<IviSurfaceWrapper*>(data);
+  // wl_output doesn't get the right information on AGL. For now,
+  // |HandleConfigure| is the only way to get the screen information.
+  // https://jira.automotivelinux.org/browse/SPEC-1940
+  // When it gets configure event from AGL, we set the size with
+  // |output_manager|.
+  WaylandOutputManager* output_manager =
+      surface->connection_->wayland_output_manager();
+  DCHECK(output_manager);
+  output_manager->HandleMetricsForPrimaryOutput(gfx::Rect(width, height));
+
```

Comment:
Isn't this a workaround? Surface size reported through configure should not match screen size (apps are not shown fullscreen in AGL window manager).

So my guess is that we are not properly processing wl_output events? Or is it weston not properly notifying us?

Reply:
We're working with wl_output events correctly with desktop shell. As discussed on https://jira.automotivelinux.org/browse/SPEC-1940, wl_output doesn't let us know the size which AGL, especially Window Manger, is using. AGL window manager can send the information only through ivi configure listener and I needed something to forward it to display. So, I can say that it's a workaround. If the window manager on AGL is improved to send the size, we can remove this solution. WDYT?

Reply:
It's soooo hacky. Do you want to say that WaylandOutput does not receive the size of the primary output when running with ivi shell?

Reply:
As you can see from https://jira.automotivelinux.org/browse/SPEC-1940, it gets the display size, not the size managed by the window manager, AGL.

Reply:
don't you just need to update WaylandScreen and pass the transform information from
https://cs.chromium.org/chromium/src/ui/ozone/platform/wayland/wayland_output.cc?dr&g=0&l=52 ?

Reply:
> don't you just need to update WaylandScreen and pass the transform information from
> https://cs.chromium.org/chromium/src/ui/ozone/platform/wayland/wayland_output.cc?dr&g=0&l=52 ?

I tried it with https://github.com/jkim-julie/chromium68/commit/d3c663bc2302490d9400f24e9e7caf84fa39a99a but it has still the issue.
It delivered output_transform information correctly to display.
I'm not sure if the management for display with rotation has the problem.

Reply:
> Could you ellaborate on how different it is the display size from the size managed in window manager? In which situations they are different and why?

I'm not quite sure about that part, either. I found the document, https://wiki.automotivelinux.org/_media/kickstart_for_wayland_client.pdf, through searching and I guessed WM works on ivi-shell from page 3.

'IviSurfaceWrapper::HandleConfigure' has '1080, 1488' and OutputHandleMode has '1920, 1080' with output_transform '3'.

So, I thought we can trust the size generated by WM through 'IviSurfaceWrapper::HandleConfigure'.

As I also think it's a workaround patch, I suggest that we just use this change only for the demo unless we find the correct solution until the demo and I'd like someone who continues to work on AGL project to work on the issue.

Reply:
The rotation is not propagated, because we don't have a support for it yet.

Check the WaylandScreen - https://cs.chromium.org/chromium/src/ui/ozone/platform/wayland/wayland_screen.cc?q=wayland_screen&dr=C&l=31. You have to pass the rotation value from the WaylandOutput and call Display::SetRotationAsDegree or Display::set_rotation (depends if you have converted the value passed from a Wayland compositor or not) - https://cs.chromium.org/chromium/src/ui/display/display.h?q=ui/display/display.h&dr&l=159

Reply:
Could you check the branch,  https://github.com/jkim-julie/chromium68/commit/d3c663bc2302490d9400f24e9e7caf84fa39a99a, I shared above? I extended the param with output_transform.

Reply:
Ok, I got it. Have you tried to see which value WL_OUTPUT_MODE_PREFERRED brings instead?

Reply:
I didn't check it recently but if you look at the logs I added at https://jira.automotivelinux.org/browse/SPEC-1940, it has 'flags:3' which means 'WL_OUTPUT_MODE_CURRENT | WL_OUTPUT_MODE_PREFERRED'.

Even if the logs are based on the external ozone port, we can refer the value from Wayland.
[1112/000015:ERROR:screen.cc(73)] OutputHandleMode, flags:3, width:1920, height:1080